h/services/group.py

```diff
@@ -62,6 +62,9 @@ def create(self, name, authority, userid, description=None, type_='private'):
         for attr, value in access_flags.iteritems():
             setattr(group, attr, value)
 
+        if group.joinable_by is not None:
+            group.members.append(creator)
```

Comment:
As I understand it `joinable_by` is something like "example.com", but there doesn't appear to be permissions code for asking "If this group joinable by this user" yet (otherwise I could have done `if group.is_joinable_by(creator)`) so I went with this. Alternatively this could also be `if type_ != publisher`.